src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_VOLUME = 10;
+    private static double INCREASING_MULTIPLICATOR = 1.5;
+    private int size = 0;
+    private Object[] list;
+
+    public ArrayList() {
+        list = new Object[DEFAULT_VOLUME];
+    }
+
+    public ArrayList(int size) {
+        list = new Object[size];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == list.length) {
+            increaseVolume((int) Math.round(list.length * INCREASING_MULTIPLICATOR));
+        }
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index != size) {
+            checkIndex(index);
+        }
+        increaseVolume(list.length + 1);
+        System.arraycopy(list, index, list, index + 1, size - index);
+        list[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) list[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Object element = list[index];
+        if (index == size - 1) {
+            list[index] = null;
+        } else {
+            System.arraycopy(list, index + 1, list, index, size - index);
+        }
+        size--;
+        return (T) element;
```

Comment:
No need casting

Reply:
The same problem. Method T can't return Object.